core/src/main/java/hivemall/optimizer/Regularization.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Hivemall: Hive scalable Machine Learning Library
+ *
+ * Copyright (C) 2015 Makoto YUI
+ * Copyright (C) 2013-2015 National Institute of Advanced Industrial Science and Technology (AIST)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package hivemall.optimizer;
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+
+public abstract class Regularization {
+
+    protected final float lambda;
+
+    public Regularization(final Map<String, String> options) {
+        float lambda = 1e-6f;
```

Comment:
I'm not sure this is the good value as default for penalty weight.
It looks a bit smaller. sklearn SGD implementation uses 0.0001 as default.
http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html